pkg/storage/storagewrappers/boundedconcurrency.go

```diff
@@ -79,7 +81,20 @@ func (b *boundedConcurrencyTupleReader) waitForLimiter(ctx context.Context) {
 
 	end := time.Now()
 	timeWaiting := end.Sub(start).Milliseconds()
-	boundedReadDelayMsHistogram.Observe(float64(timeWaiting))
+
+	method, found := grpc.Method(ctx)
+	if !found {
+		method = "undefined"
+	}
+
+	// grpc Method returns in the format "/service/method".
+	methods := strings.Split(method, "/")
+
+	boundedReadDelayMsHistogram.WithLabelValues(
+		openfgav1.OpenFGAService_ServiceDesc.ServiceName,
+		methods[len(methods)-1],
+	).Observe(float64(timeWaiting))
+
```

Comment:
```suggestion

        // grpc.Method returns in the format "/service/method". 
	serviceAndMethod, found := grpc.Method(ctx)
	if !found {
		method = "undefined"
	} else {
          	method = strings.Split(serviceAndMethod, "/")[2]
	}
	boundedReadDelayMsHistogram.WithLabelValues(
		openfgav1.OpenFGAService_ServiceDesc.ServiceName,
		method,
	).Observe(float64(timeWaiting))

```

Reply:
This is why I use len(methods)-1 as it will not panic if the method "/" is not as expected.

Reply:
Also, I believe @jon-whit does not want the grpc.Method to be used here. One thought is to add to context "Method" in the server method (i.e., Check/ListObject) and retrieve it here.  Want to hear your thought on that @miparnisari

Reply:
> Also, I believe @jon-whit does not want the grpc.Method to be used here.

why?

Reply:
It leaks transport specific details deeply in the storage layer. This should at minimum be moved down a level as a contract between the app layer and the storage layer, not the transport layer and the storage layer. e.g.

```go
// pkg/server/server.go

package server

func (s *Server) Check(ctx context.Context, req *openfgav1.CheckRequest) (*openfgav1.CheckResponse, error) {
    ctx = ContextWithRPCMethod(fmt.Sprintf("%s/%s", openfgav1.OpenFGAService_ServiceDesc.ServiceName, "Check"

    ...
}
```
then in the storage layer
```go
// pkg/storage/storagewrappers/boundedconcurrency.go

package storagewrappers

func (b *boundedConcurrencyTupleReader) waitForLimiter(ctx context.Context) {
   ...
   method := RPCMethodFromContext(ctx)

   _ = method // use the method in the histogram observation
}
```

Reply:
I am ok with moving it up - but just want to get confirmation that we are good with using context to pass these info.

Reply:
@miparnisari passing values through context has been met with resistance in the past. I think this is a good use of context value propagation, but what's your take?

Reply:
I took a stab and updated to pass through context.  @jon-whit / @miparnisari - please raise your voice if there are any objections.

Reply:
We talked about not passing it through the transport layer, and we added more code and are still passing it through the transport layer. If we're going to extract context that is injected at the transport layer, then let's just go back to using `grpc.Method(ctx)`, as that is more standard and less code we have to manage. Otherwise, please move this to the application layer. Middleware interceptors are gRPC transport specific, so what we're doing right now is no different than what we had IMO. We already get the method context injected into the ctx from the grpc middleware, so no need to duplicate that work if we're going to inject that value at the transport middleware layer.

The suggestion above was to move the method context injection into the app layer directly, which is transport agnostic.

Reply:
As discussed offline, we will go back to adding the method via method in the app layer